segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -314,15 +313,17 @@ private ReadResultEntry collectCachedEntries(long initialOffset, ReadResult read
     /**
      * Collect all the data from the given contents into a {@link ByteBuf}.
      */
-    private ByteBuf getData(List<BufferView> contents) {
-        val compositeView = BufferView.wrap(contents);
-        val rawBuffers = compositeView.getContents();
-        val result = Unpooled.compositeBuffer(rawBuffers.size());
-        for (ByteBuffer b : rawBuffers) {
-            result.addComponent(Unpooled.wrappedBuffer(b));
-        }
+    private ByteBuf toByteBuf(List<BufferView> contents) {
+        val buffers = contents.stream()
```

Comment:
In our experiments, we normally observe that using `stream()` heavily is more costly than a plain loop. We need to evaluate whether this change is making any performance difference in this case.

Reply:
Sure. But in this case I need to output a byte array from a list of lists, so using `Stream.flatMap` helps a lot. There isn't any more efficient way of doing this and I would just be writing a lot more code to get it done.